### PR TITLE
135 cablenml improvements

### LINKED ISSUE #135: cable.nml improvements
The cable.nml documentation can be improved.

- Many entries in the description table do not actually have default values.
- Some default values are wrong. Eg. `vegparmnew = .FALSE.`
- Some namelist variables are missing.
- Why is the default `restart_in = ' '` ? What does it mean?
- The table widths could be adjusted.
- It looks like filename%veg and filename%soil are not used.

### PR DESCRIPTION
**Pull request by tammasloughran**
# CABLE

Thank you for submitting a pull request to the CABLE Project.

## Description

The cable.nml documentation can be improved.

- Many entries in the description table do not actually have default values.
- Some default values are wrong. Eg. vegparmnew = .FALSE.
- Some namelist variables are missing.
- Why is the default restart_in = ' ' ? What does it mean?
- The table widths could be adjusted.
- It looks like filename%veg and filename%soil are not used.

Fixes #135

## Type of change

Please delete options that are not relevant.

- [x] New or updated documentation

## Checklist

- [x] The new content is accessible and located in the appropriate section.
- [x] I have checked that links are valid and point to the intended content.
- [x] I have checked my code/text and corrected any misspellings

Please add a reviewer when ready for review.

----
:books: Documentation preview :books:: https://cable--136.org.readthedocs.build/en/136/

### CODE REVIEW

**Comment by tammasloughran** [comment]
I think I'm at the point where I need others to fill in the missing descriptions.
Should the namelist variables related to POP stay here? or is there a separate location for POP related documentation?

**Comment by JhanSrbinovsky** [comment]
I suppose cablenml.md should reflect cable.nml, which is in desperate need of revision. At least in the MD we can organise things into sections which will eventually be namelists in their own right. For e.g. Anything to do with filename% and output%  doesn't belong in a generic cable.nml. The UM doesn't use any of it. We should have casa.nml and pop.nml (etc etc) files. You're right to get rid of the filename%veg/soil.

**Comment by ccarouge** [comment]
I agree we should put some order into this but this PR was to look at the correctness of the information. I think there is already enough done here and don't need to add. I will put your comment @JhanSrbinovsky as an issue so we can discuss there how to organise the documentation for the namelist.

**Review by ccarouge — CHANGES_REQUESTED** [review]
A few changes, especially some duplicated entries. But otherwise it's good work.

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 1]
```diff
@@ -83,74 +83,138 @@ applications. The following are annotated examples of cable.nml:
...
+| filename%path                    | character(len=500) | any string of max. 500 characters                          | './'                             | Path for output and restart files for CABLE and CASA.                                                   |
+| filename%out                     | character(len=500) | any string of max. 500 characters                          | uninitialised                    | Output file name.                                                                                       |
+| filename%log                     | character(len=500) | any string of max. 500 characters                          | uninitialised                    | Execution log file name.                                                                                |
+| filename%restart_in              | character(len=500) | any string of max. 500 characters                          | ' '                              | Input restart file name. '' means curent directory.                                                     |
```
[the remark this answers is not in the document]
Comment (tammasloughran) [reply documentation/docs/user_guide/inputs/cable_nml.md:92]: I think both are valid descriptions. I had a look at `cable_input.F90` and it seems when this is `''`, it first searches for restart files in the current directory named `./<RunIden>_<year>_cable_rst.nc`, then it uses default parameters.

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 2]
```diff
@@ -83,74 +83,138 @@ applications. The following are annotated examples of cable.nml:
...
+| output%veg                       | logical            | .TRUE. .FALSE.                                             | .FALSE.                          | Output vegetation states when .TRUE.                                                                    |
+| output%params                    | logical            | .TRUE. .FALSE.                                             | .FALSE.                          | Output the input parameters used to produce run then .TRUE.                                             |
+| output%balances                  | logical            | .TRUE. .FALSE.                                             | .FALSE.                          | Output energy and water balances when .TRUE.                                                            |
+| output%grid                      | character(len=7)   | 'default' 'land' 'mask' 'ALMA'                             | 'default'                        | Output grid. 'default is the same as the met file'.                                                     |
```
[the remark this answers is not in the document]
Comment (tammasloughran) [reply documentation/docs/user_guide/inputs/cable_nml.md:118]: Not sure. All the other values need more explanation too. Just now checked the `cable_output.F90` file to see what this means. I think it's outputting 1D land-only array or 2D masked spatial arrays. I couldn't figure out what ALMA is from the code. Google searches tell me that ALMA is a file convention called Assistance for Land-surface Modelling Activities... it think?
So I suppose `'default'` means that the output files use whatever convention the input meterology file is using.

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 3]
```diff
@@ -83,74 +83,138 @@ applications. The following are annotated examples of cable.nml:
...
+| check%energy_bal                 | logical            | .TRUE. .FALSE.                                             | uninitialised                    | Energy balance                                                                                          |
+| check%mass_bal                   | logical            | .TRUE. .FALSE.                                             | uninttialised                    | Water/mass balance                                                                                      |
+| verbose                          | logical            | .TRUE. .FALSE.                                             | uninitialised                    | Write details of every grid cell initialisation and parameters to log.                                  |
+| leaps                            | logical            | .TRUE. .FALSE.                                             | depends on configuration         | Calculate timing with leap years?                                                                       |
```
[the remark this answers is not in the document]
Comment (tammasloughran) [reply documentation/docs/user_guide/inputs/cable_nml.md:124]: I probably did have some question but I've forgotten.

**Comment by tammasloughran** [comment]
I'm having difficulty understanding what `cable_user%soil_thermal_fix` is. Does anyone know what this is?

**Comment by JhanSrbinovsky** [comment]
> I'm having difficulty understanding what `cable_user%soil_thermal_fix` is. Does anyone know what this is?

From memory an alternative soil conductivity implementation from Mark Deker - the Group Water guy. I did think we were using it in CM2, but it doesn't appear so.

**Comment by tammasloughran** [comment]
I don't know what `cable_user%initialize_mapping` is either. @JhanSrbinovsky , it looks like you wrote this?

**Comment by tammasloughran** [comment]
`cable_user%cable_runtime_coupled` seems to only do some snow stuff when uncoupled. Why and What?

**Comment by JhanSrbinovsky** [comment]
> I don't know what `cable_user%initialize_mapping` is either. @JhanSrbinovsky , it looks like you wrote this?

You can kill this. It is a relic from saving the ACCESS forcing. It has been (potentially will be again) as a diagnostic tool. But in no way needed for the model.

**Comment by JhanSrbinovsky** [comment]
> `cable_user%cable_runtime_coupled` seems to only do some snow stuff when uncoupled. Why and What?

This one is still in the code, largely redundant yet requires further thought. We flipped flopped over the default for a while and it's one of those things that I have to start all over again to understand and remove it. It is a relic from CMIP5 which initially started from a dead cold state, i.e. tiled soils etc didn't exist in the UM, had no values, let alone spun up. Then it morphed as CMIP5 UM didn't keep track the date and we needed to dodge it (we had to have a post-processing date tracker - it was crazy). Then it resurfaced in CMIP6 with a sightly different usage requirement again. Gives me a headache just thinking about it. Kill it and maybe @ccarouge can flag it?

**Comment by tammasloughran** [comment]
Many thanks! The only other entries I can't make sens of are the GW related flags.

- `cable_user%litter`
- `cable_user%gw_model`
- `cable_user%or_evap`
- `cable_user%test_new_gw`

Can someone that knows these well can describe them for me?

**Comment by ccarouge** [comment]
@JhanSrbinovsky I'm not certain what you mean by "kill it" and "flag it". I would assume "kill it" means remove it? But "flag it", I have no idea.

**Comment by JhanSrbinovsky** [comment]
I think Tammas has  not bothered with documentation of it (the runtime_coupled flag) as it is basically redundant. And by flag it I mean keep note of it as something to remove from the code (at a later date is fine. Its not breaking anything)
Jhan

From: Claire Carouge ***@***.***>
Date: Friday, 14 July 2023 at 4:19 pm
To: CABLE-LSM/CABLE ***@***.***>
Cc: Srbinovsky, Jhan (Environment, Aspendale) ***@***.***>, Mention ***@***.***>
Subject: Re: [CABLE-LSM/CABLE] 135 cablenml improvements (PR #136)

@JhanSrbinovsky<https://github.com/JhanSrbinovsky> I'm not certain what you mean by "kill it" and "flag it". I would assume "kill it" means remove it? But "flag it", I have no idea.

**Comment by ccarouge** [comment]
OK. So I've added an issue in this repository (#137 ) to add the deprecated namelist options to the list Ramzi has started in the User Guide. We can add to that issue as we find other such options.

**Comment by ccarouge** [comment]
> - cable_user%litter

All I know is that it adds a litter layer on top of the soil surface. It was implemented because CABLE was/is over-evaporating and this reduces the evaporation. But that might not be useful information for the User Guide.

> - cable_user%gw_model

This is to turn on an alternate soil model that takes into account the groundwater. @bibivking How best would you describe this in the documentation when describing the namelist options?

> - cable_user%or_evap

This is an evaporation scheme from Dani Or. But I don't know if there is a reference for it. @bibivking Do you know more about this scheme?

> - cable_user%test_new_gw

I would think this one will disappear when Ramzi is finished merging the GW scheme. The technical phase should be finished by the end of August, so I would put this in deprecated and not list it here. @rkutteh any thoughts on this namelist option? Have you encountered it so far in your work merging the versions?

**Comment by rkutteh** [comment]
cable_user%litter
cable_user%gw_model
cable_user%or_evap
cable_user%test_new_gw

Of the four above, the only one I have not encountered yet is the last one (but I am midway through the work still...). The second and third are definitely in use and needed. I will get back to this thread in a month or so when I am finished with moving the GW into the trunk and say more...

**Comment by tammasloughran** [comment]
I have updated everything with the information provided, thanks folks! It looks like all the namelist variables here have some sort of description.
Should there be an order? Alphabetical? The most important ones up the top? Or is it fine as it is?

**Comment by ccarouge** [comment]
@tammasloughran I don't think we want to worry about an order now. We will want to split this huge namelist into several namelists soon. The ordering would happen then. The webpage rendered is static so no ordering possible there.

**Comment by ccarouge** [comment]
@tammasloughran Is it ready for another review? If so, can you request it? Either look at the "Reviewers" menu on the top right and click the "2 arrows in a circle" icon, a bit like this: 🔄
Or at the bottom of the page, where it says "1 change requested", expand that portion, click on the "3 dots icon" and choose "Re-request review".

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 4]
```diff
@@ -83,74 +83,127 @@ applications. The following are annotated examples of cable.nml:
...
+| filename%restart_out             | character(len=500) | any string of max. 500 characters                          | uninitialised                    | Output restart file name.                                                                               |
+| filename%type                    | character(len=500) | any string of max. 500 characters                          | uninitialised                    | Static information file about grid, soil and vegetation.                                                |
+| filename%lai                     | character(len=500) | any string of max. 500 characters                          | uninitialised                    | Default leaf area index file name.                                                                      |
+| filename%soilcolor               | character(len=500) | any string of max. 500 characters                          | uninitialised                    | Soil color (soilcolor_global_1x1.nc) file name.                                                         |
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:96]:
```suggestion
| filename%soilcolor               | character(len=500) | any string of max. 500 characters                          | uninitialised                    | Soil color file name. E.g. "CABLE-AUX/offline/soilcolor_global_1x1.nc"             |
```

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 5]
```diff
@@ -83,74 +83,127 @@ applications. The following are annotated examples of cable.nml:
...
+| delsoilT                         | real               | any real number                                            | uninitialised                    | Allowed variation in soil temperature for spin up.                                                      |
+| output%restart                   | logical            | .TRUE. .FALSE.                                             | uninitialised                    | Create a restart file when .TRUE..                                                                      |
+| output%met                       | logical            | .TRUE. .FALSE.                                             | .FALSE.                          | Output input meteorological data when .TRUE..                                                           |
+| output%flux                      | logical            | .TRUE. .FALSE.                                             | .FALSE.                          | Output convective, runoff, NEE.                                                                         |
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:108]:
```suggestion
| output%flux                      | logical            | .TRUE. .FALSE.                                             | .FALSE.                          | Output convective, runoff, NEE when TRUE                                                                |
```

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 6]
```diff
@@ -83,74 +83,127 @@ applications. The following are annotated examples of cable.nml:
...
+| cable_user%POP_out               | character(len=3)   | 'epi' 'rst' 'ini'                                          | 'rst'                            | POP restart file type. `'epi'` is end of year state `'rst'` is a standard restart file, `'ini'` is an initialisation restart file. |
+| cable_user%POP_rst               | character(len=50)  | any string of max. 50 characters                           | ' '                              | POP restart file directory.                                                                             |
+| cable_user%casa_out_freq         | character(len=8)   | 'daily' 'monthly' 'annually'                               | 'annually'                       | CASA output frequency.                                                                                  |
+| cable_user%vcmax                 | character(len=10)  | 'standard' 'Walker2014'                                    | 'standard'                       | Alternative functional form of $V_{cmax}$ (maximum RuBP carboxylation rate). For `'Walker2014'` see doi:10.1002/ece3.1173 . |
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:171]:
```suggestion
| cable_user%vcmax                 | character(len=10)  | 'standard' 'Walker2014'                                    | 'standard'                       | Alternative functional form of $V_{cmax}$ (maximum RuBP carboxylation rate). For `'Walker2014'` see [Walker et al., 2014](https://doi.org/10.1002/ece3.1173) |
```

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 7]
```diff
@@ -83,74 +83,127 @@ applications. The following are annotated examples of cable.nml:
...
-| cable_user%CASA_DUMP_READ    | logical   | .TRUE. .FALSE.                      | .FALSE.                                                                      | TRUE reads CASA forcing from netcdf format                                                               |
-| cable_user%CASA_DUMP_WRITE   | logical   | .TRUE. .FALSE.                      | .FALSE.                                                                      | TRUE outputs CASA forcing in netcdf format                                                               |
-| cable_user%SSNOW_POTEV       | character(len=3) | 'HDM' 'P-M'                  | 'HDM'                                                                        | Use the "Humidity Deficit Method" or  "Penman-Monteith" method                                           |
+| Namelist variable                | Type               | Available values                                           | Default value                    | Description                                                                                             |
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:86]:
```suggestion
## List of namelist variables

| Namelist variable                | Type               | Available values                                           | Default value                    | Description                                                                                             |
```

**Review by ccarouge — APPROVED** [review]
I have minor suggestions to make things a bit clearer.

I would prefer if the table wasn't between the example for offline and ACCESS simulations. So if you can move either the table first and the examples after or the other way around that would be great.

But I have approved as is so feel free to merge in when you think it is ready.

File: documentation/docs/user_guide/inputs/cable_nml.md (hunk 7 above)
Comment (ccarouge) [reply documentation/docs/user_guide/inputs/cable_nml.md:86]: This appears weird here but correct in the review panel. This should simply add a section title before the table.

**Comment by tammasloughran** [comment]
I'm not sure what to do about the link checks. At first I thought there was some problem with the DOI redirection but changing it to a full URI didn't make any difference.

**Comment by ccarouge** [comment]
It's possible to ignore specific links in the link checker: https://github.com/gaurav-nelson/github-action-markdown-link-check#disable-check-for-some-links . So I've added this for the link to the paper. I've changed the link to go via doi.org so it will stay valid even if the journal is sold.

**Pull request merged**